### PR TITLE
added emoji feature

### PR DESCRIPTION
**Pull request by geetanjalidubey9**
install the required dependency 

yarn add emoji-picker-react

### CODE REVIEW

File: frontend/src/components/SingleChat.js [hunk 1]
```diff
@@ -233,6 +245,15 @@ const SingleChat = ({ fetchAgain, setFetchAgain }) => {
                   />
                 </div>
               ) : null}
+<Button onClick={toggleEmojiPicker} ml={2}>
+  <BsEmojiSmile size={24} /> 
+</Button>
+              {showEmojiPicker && (
+                <div style={{ position: "absolute", bottom: "60px", zIndex: 100 }}>
```
Comment (kartikayasijaa) [inline frontend/src/components/SingleChat.js:252]: Can you fix the intendation?

File: frontend/src/components/SingleChat.js [hunk 2]
```diff
@@ -100,6 +104,14 @@ const SingleChat = ({ fetchAgain, setFetchAgain }) => {
       }
     }
   };
+const handleEmojiClick = (emojiObject) => {
+  setNewMessage(newMessage + emojiObject.emoji); // Correct usage to append emoji
+};
+
+const toggleEmojiPicker = () => {
+  setShowEmojiPicker(!showEmojiPicker); // Toggle the visibility of emoji picker
```
Comment (kartikayasijaa) [inline frontend/src/components/SingleChat.js:112]: can you do like this `setShowEmojiPicker(emoji => !emoji)`

**Comment by kartikayasijaa** [comment]
@geetanjalidubey9 I have added comments, can you check?

File: frontend/src/components/SingleChat.js (hunk 2 above)
Comment (geetanjalidubey9) [reply frontend/src/components/SingleChat.js:112]: okay let me correct it

**Comment by geetanjalidubey9** [comment]
okay

**Comment by geetanjalidubey9** [comment]
https://github.com/user-attachments/assets/8f60bf3c-b2f0-45ed-bd8d-800281c18d92

**Comment by geetanjalidubey9** [comment]
pls check it and let me knw

**Comment by kartikayasijaa** [comment]
closes #33

**Pull request merged**